Adding new context method local_md5  for v1.4 (#2658)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/local-md5.md

---
title: "local_md5"
id: "local_md5"
---

> The `local_md5` context variable calculates an [MD5 hash](https://en.wikipedia.org/wiki/MD5) of the given string. It's called `local_md5` to emphasize that the hash is calculated _locally_, in the dbt-Jinja context. This function is intended for generally advanced use cases: generating unique identifiers within custom materialization or operational logic, such as to avoid collisions between temporary relations, or to identify changes by comparing checksums.

Reviewer: ```suggestion
The `local_md5` context variable calculates an [MD5 hash](https://en.wikipedia.org/wiki/MD5) of the given string. The name, `local_md5`, emphasizes that the hash is calculated _locally_, in the dbt-Jinja context. You should use this function for generally advanced use cases. For example, when you generate unique identifiers within custom materialization or operational logic, which helps you either avoid collisions between temporary relations or identify changes by comparing checksums.
```

> The `local_md5` context variable calculates an [MD5 hash](https://en.wikipedia.org/wiki/MD5) of the given string. The name, `local_md5`, emphasizes that the hash is calculated _locally_, in the dbt-Jinja context. You should use this function for generally advanced use cases. For example, when you generate unique identifiers within custom materialization or operational logic, which helps you either avoid collisions between temporary relations or identify changes by comparing checksums.

Reviewer 2: ```suggestion
The `local_md5` context variable calculates an [MD5 hash](https://en.wikipedia.org/wiki/MD5) of the given string. The string `local_md5` emphasizes that the hash is calculated _locally_, in the dbt-Jinja context. This variable is typically useful for advanced use cases. For example, when you generate unique identifiers within custom materialization or operational logic, you can either avoid collisions between temporary relations or identify changes by comparing checksums.
```

Reviewer 2: sugg for removing "should" since that's harder to translate/localize

Author: Good catch!

> It should not be confused with the `md5` SQL function, supported by many SQL dialects, which runs remotely in the data platform. You should always use SQL hashing functions when generating <Term id="surrogate-key">surrogate keys</Term>.

Reviewer 2: ```suggestion
It is different than the `md5` SQL function, supported by many SQL dialects, which runs remotely in the data platform. You want to always use SQL hashing functions when generating <Term id="surrogate-key">surrogate keys</Term>.
```

Reviewer 2: sugg: alternatives to "should"
